File: 169-majority-element/majority-element.go
func majorityElement(nums []int) int {
    appearence := make(map[int]int)
    majority := len(nums) / 2
    for i := 0; i < len(nums); i++{
        appearence[nums[i]]++
    } 
    for i := 0; i < len(nums); i++{
        if appearence[nums[i]] > majority{
            return nums[i]
        }
    } 

    return 0
}

File: 169-majority-element/majority-element.py
class Solution(object):
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        count = 0
        selectedNum = None

        for num in nums:
            if count == 0:
                selectedNum = num
            count += 1 if num == selectedNum else -1

        return selectedNum
        